Makefile

```diff
@@ -1,28 +1,44 @@
-# Select architecture: RV32 or RV64.
-ARCH ?= RV64
+# Select architecture
+ifeq ($(ARCH),RV32)
+  override XLEN := 32
+  override FLEN := 32
```

Comment:
In the original Makefile, it was only possible to specify either RV32 which would result in XLEN=FLEN=32, or RV64 which would result in  XLEN=FLEN=64.  My modifications allow for backward compatibility, thus if the same value of ARCH was passed to the Makefile it would result in the same behavior, but it also allows for the FLEN and XLEN values to be specified explicitly as 64 or 32 to allow for the various other cases.